contracts/axelarnet-gateway/src/client.rs

```diff
@@ -0,0 +1,131 @@
+use cosmwasm_std::{HexBinary, WasmMsg};
+use router_api::{Address, ChainName, CrossChainId, Message};
+
+use crate::msg::{ExecuteMsg, QueryMsg};
+
+impl<'a> From<client::Client<'a, ExecuteMsg, QueryMsg>> for Client<'a> {
+    fn from(client: client::Client<'a, ExecuteMsg, QueryMsg>) -> Self {
+        Client { client }
+    }
+}
+
+pub struct Client<'a> {
+    client: client::Client<'a, ExecuteMsg, QueryMsg>,
+}
+
+impl<'a> Client<'a> {
+    pub fn call_contract(
+        &self,
+        destination_chain: ChainName,
+        destination_address: Address,
+        payload: HexBinary,
+    ) -> WasmMsg {
+        self.client.execute(&ExecuteMsg::CallContract {
+            destination_chain,
+            destination_address,
+            payload,
+        })
+    }
+
+    pub fn execute(&self, cc_id: CrossChainId, payload: HexBinary) -> WasmMsg {
+        self.client.execute(&ExecuteMsg::Execute { cc_id, payload })
+    }
+
+    pub fn route_messages(&self, msgs: Vec<Message>) -> Option<WasmMsg> {
+        ignore_empty(msgs).map(|messages| self.client.execute(&ExecuteMsg::RouteMessages(messages)))
+    }
+}
+
+fn ignore_empty(msgs: Vec<Message>) -> Option<Vec<Message>> {
```

Comment:
it can be extracted to package

Reply:
the whole client, or just the helper?

Reply:
just the helper is good for now, we redefined the same fn multiple times already